std/encoding/name_pattern.go

```diff
@@ -104,21 +104,21 @@ func (n Name) Hash() uint64 {
 	h := hashPool.Get().(hash.Hash64)
 	defer hashPool.Put(h)
 	h.Reset()
-	for _, c := range n {
-		c.HashInto(h)
+	for i := range n {
+		n[i].HashInto(h)
 	}
 	return h.Sum64()
 }
 
 // PrefixHash returns the hash value of all prefixes of the name
 // ret[n] means the hash of the prefix of length n. ret[0] is the same for all names.
-func (n Name) PrefixHash() []uint64 {
+func (n *Name) PrefixHash() []uint64 {
```

Comment:
Why did you change it to a pointer receiver?

- Value receiver is strongly preferred for slices unless the function modifies the slice itself.
- Generally, two types of receivers should not be mixed to avoid confusion.

See: https://google.github.io/styleguide/go/decisions#receiver-type

Reply:
Oops, this was supposed to be reverted. Done now 👍🏻 

Context: Component::HashInto (earlier) allocated a slice for the TLV type, which was passed to `hash.Hash`. Because `hash.Hash`, this slice escaped to heap, even with `unsafe.Pointer`. The only way to prevent a copy was to already have a pointer to `Component` in `HashInto`. For this method, this meant that if `n` was on stack, `c` would be copied to heap first. 

But now, a double pointer redirection in `HashInto` ensures that we directly use the stack address without a copy, so all this is no longer required.